### ISSUE TITLE
Notes de bas de page et blocs spéciaux

### ISSUE THREAD

**Issue by R-Alexandre** [R-Alexandre/ClassesTex#1]
# Le problème 

Actuellement le code

`
\definition{\note{Hello}}{}
`

provoque une erreur. Sa gestion par : 

`
\definition{\footnotemark}{}\footnotetext{Hello}
`

peut empêcher son positionnement sur la bonne page lorsque le bloc est à cheval sur plusieurs pages.

# Dans l'idéal

Trouver un correctif sans modifier la nature du bloc.

# Sinon

Changer la nature de ces blocs, quitte à changer (un peu, mais pas trop : il faudra en juger) l'esthétique, pour permettre de telles notes de bas de page.

**Comment by R-Alexandre** [comment]
Avec la dernière PR, on peut maintenant utiliser `\note{}` partout.

**Comment by Teguad** [comment]
Oui, mais le placement n'est pas encore correcte.

J'ai créé un sujet sur [TeX Stackexchange](http://tex.stackexchange.com/q/291663). En attendant des réponses qui peuvent régler ton problème... (Et 'scuse moi pour mon anglais.) :stuck_out_tongue:

**Comment by R-Alexandre** [comment]
Il me semble qu'il y a un bug de numérotation en bas de page quand plusieurs `\note{}` interviennent dans un même bloc, non ?

**Comment by Teguad** [comment]
Je pense avoir réglé le problème dans la PR #5.

**Comment by R-Alexandre** [comment]
Il y a un nouveau bug. Au changement d'un chapitre la dernière note de bas de page peut se répéter : 
[cours.pdf](https://github.com/R-Alexandre/ClassesTex/files/123836/cours.pdf)

**Comment by R-Alexandre** [comment]
Problème résolu en retirant le correctif apporté en #4

**Comment by Teguad** [comment]
Ma solution n'était pas optimale alors. Vaut mieux faire les notes "à la main". :-)

**Comment by R-Alexandre** [comment]
Sinon j'ai peut-être une piste pour régler ce problème.

Ça serait de changer la structure de ces blocs. Au lieu d'avoir une barre sur le côté (qui demande l'utilisation d'un environnement spécial) je propose de revenir au format "classique" mais sous la forme suivante : 

- garder le même entête du style "Définition (Espace vectoriel)"
-  avoir un indicateur de fin d'énoncé : \cqfd pour une preuve, \blacklozenge  pour une définition et \spadesuit pour un proposition/théorème/lemme et le reste on verra plus tard.

Est-ce que ça te semble réalisable ? Est-ce que ça permettrait de résoudre le problème de notes de bas de page ?

**Comment by Teguad** [comment]
Oui, c'est une bonne idée et ça réglerai tous les problèmes. C'est facilement réalisable en plus.

Je ferrai ça par contre ce week-end, je suis mort. :-)

**Comment by Teguad** [comment]
Salut Holosmos !

Désolé du retard, mon week-end est très chargé.

J'ai codé ce que tu m'as demandé. J'en ai profité pour simplifier la création des théorèmes, tu verras. :-)
Donc, voici la nouvelle définition des théorèmes.
```latex
% --- Encadrement ---

\newif\iffin\finfalse
\newcommand{\thmEncadre}[2]{%
	\expandafter\def\csname fin#1\endcsname{%
		\global\fintrue
		\ifmmode\tag*{#2}\else\hfill#2\fi
	}
	\long\expandafter\def\csname #1\endcsname##1##2{
		\def\fin{\expandafter\csname fin#1\endcsname}
		\global\finfalse
		\expandafter\csname #1T\endcsname[##2]%
			##1
			\iffin\else%
				\expandafter\csname fin#1\endcsname%
				\global\finfalse%
			\fi 
		\expandafter\csname end#1T\endcsname
	}
}

% --- Définition des théorèmes ---

\newtheorem{definitionT}{Définition}[section]
\newtheorem{propositionT}[definitionT]{Proposition}
\newtheorem{theoremeT}[definitionT]{Théorème}
\newtheorem{loiT}[definitionT]{Loi}
\newtheorem*{demonstrationT}{Preuve}
\newtheorem*{preuveT}{Preuve}
\newtheorem{axiomeT}[definitionT]{Axiome}
\newtheorem{lemmeT}[definitionT]{Lemme}
\newtheorem{corollaireT}[definitionT]{Corollaire}
\newtheorem{procedeT}[definitionT]{Procédé}

% --- Théorèmes encadrés ---

\thmEncadre{definition}{$\blacklozenge$}
\thmEncadre{proposition}{$\spadesuit$}
\thmEncadre{theoreme}{$\spadesuit$}
\thmEncadre{loi}{$\blacklozenge$}
\thmEncadre{demonstration}{$\square$}
\thmEncadre{preuve}{$\square$}
\thmEncadre{axiome}{$\blacklozenge$}
\thmEncadre{lemme}{$\diamond$}
\thmEncadre{corollaire}{$\diamond$}
\thmEncadre{procede}{$---$}
```
J'ai mis des symboles arbitraires, libre à toi de les changer (second argument de `\thmEncadre`). Si tu veux en créer de nouveaux, il faut juste faire attention à bien mettre un `T` à la fin du théorème et de le déclarer avec `\thmEncadre` sans le `T`.

Après, les symboles vont se mettre automatiquement sauf si tu finis par une formule math-*display*  ou une liste auquel cas tu dois ajouter une commande `\fin` comme ceci.
```latex
\documentclass{livre}

\titre{Test}

\begin{document}

\definition{Bla}{Moi}

\definition{t
	\[ AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA \fin \]
}{Moi}

\end{document}
```

Autre chose : j'ai vu que tu avais mis des triangles pour les paragraphes. Si tu veux que je les place dans la marge, je peux.

**Comment by R-Alexandre** [comment]
Après discussion avec un ami, ce serait bien de pouvoir garder une option pour sélectionner des résultats (ou preuves) qui garderaient un trait de côté. Est-ce possible ?

Est-ce que tu pourrais aussi me faire une PR ainsi qu'un petit pdf de prévisualisation ? Merci :-).

**Comment by Teguad** [comment]
Euh oui, c'est possible de faire un autre environnement avec un trait. L'environnement s'appellerait comment ?

OK pour la PR !

**Comment by R-Alexandre** [comment]
Humm, quelque chose comme \definitionT ?

**Comment by Teguad** [comment]
OK, par contre, tu veux que cet environnement incrémente la numérotation de ceux sans trait ou pas ?

**Comment by R-Alexandre** [comment]
Oui ! Dans l'idéal, il faudrait pas voir d'autre différence que la barre entre les deux environnements.
